src/main/java/org/jboss/windup/operator/controller/WindupController.java

```diff
@@ -0,0 +1,80 @@
+package org.jboss.windup.operator.controller;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import lombok.extern.java.Log;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.windup.operator.model.WindupResource;
+import org.jboss.windup.operator.model.WindupResourceDoneable;
+import org.jboss.windup.operator.model.WindupResourceList;
+import org.jboss.windup.operator.util.WindupDeployment;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Log
+@ApplicationScoped
+public class WindupController implements Watcher<WindupResource> {
```

Comment:
_Comment to further review the code, NOT mandatory for this PR_
I think naming this `WindupWatcher` would be better since it's a watcher.
```suggestion
public class WindupWatcher implements Watcher<WindupResource> {
```

Reply:
I prefer naming by the function not by the exact implementation or extension